src/tests/stdlib/memcpy.c

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file
+ *
+ * @date Mar 22, 2020
+ * @author: Athreya Nagaraj
+ */
+
+#include <string.h>
+#include <embox/test.h>
+#define ARR_SIZE 5
+
+EMBOX_TEST_SUITE("memcpy test suit");
+
+TEST_CASE("memcpy with zero size") {
+	char src[] = "string";
+	char dst[sizeof(src) + 1];
+
+	test_assert_equal(strlen(src), strlen(memcpy(dst, src, 0)));
+}
+
+TEST_CASE("copy string with memcpy") {
+	char src[] = "string";
+	char dst[sizeof(src) + 1];
+
+	memset(dst, 0x55, sizeof(dst));
+
+	test_assert_equal(strlen(src), strlen(memcpy(dst, src, sizeof(dst))));
+	test_assert_zero(strcmp(dst, src));
+}
+
+TEST_CASE("copy integer array with memcpy") {
+	int src[] = {1, 2, 3, 4, 5};
+	int dst[ARR_SIZE * sizeof(int)];
+
+	memset(dst, 0x55, sizeof(dst));
+	memcpy(dst, src, ARR_SIZE * sizeof(int));
```

Comment:
sizeof(dst) == sizeof(int) * (sizeof(int) * ARR_SIZE)
Why do you want to use ARR_SIZE at all?